crawl-ref/source/command.cc

```diff
@@ -796,14 +796,13 @@ static void _add_insert_commands(column_composer &cols, const int column,
 }
 
 static void _add_insert_commands(column_composer &cols, const int column,
-                                 const string desc, const int first, ...)
+                                 const string desc, ...)
```

Comment:
This was a very good find -- it's bad enough that |amethyst went ahead and implemented a much more drastic solution in 022a3cd6c2ec4970f0af811bda2f5684c8d63321 that eliminates the use of varargs altogether.  (As far as we can tell, nargs is a preemptive defense against the fact that it's very easy to use these versions of _add_insert_commands wrong and not terminate the varargs lists, in fact in a bunch of cases it was doing so.). There's lots more detailed discussion in the devlog from 10/16.

Reply:
My preferred protection against not terminating varargs in this case would be GCC's `__attribute__ ((sentinel))`, but in C++ NULL != 0, so it caused a bunch of warnings when I tried it.

Reply:
Yeah, I think from the c++ perspective any use of varargs is considered not great, so |amethyst just converted it to a vector.

Reply:
Ah, found the IRC logs, which for some reason my IRC bouncer isn't working properly... I don't remember this causing a crash, I think it just popped up in valgrind while I was trying to track down one of the other resize issues. But in any case, glad it's fixed :)